python/plugins/processing/algs/gdal/extractprojection.py

```diff
@@ -73,13 +73,15 @@ def getConsoleCommands(self, parameters, context, feedback, executing=True):
         return [self.commandName()]
 
     def processAlgorithm(self, parameters, context, feedback):
-        rasterPath = self.getParameterValue(self.INPUT)
-        createPrj = self.getParameterValue(self.PRJ_FILE)
-
-        raster = gdal.Open(str(rasterPath))
-        crs = raster.GetProjection()
-        geotransform = raster.GetGeoTransform()
+        createPrj = QgsProcessingParameterBoolean(self.PRJ_FILE)
+        raster = self.parameterAsRasterLayer(parameters, self.INPUT, context)
+        rasterPath = raster.source()
```

Comment:
Here you should double check:

    if not raster.dataProvider().name() == 'gdal':
        raise QgsProcessingException('This algorithm can only be used with GDAL raster layers')

Reply:
Added (and now event with tr)